users/tests.py

```diff
@@ -0,0 +1,96 @@
+from django.test import TestCase
+from django.contrib.auth import get_user_model
+from django.db import transaction, IntegrityError
+
+
+class UserTests(TestCase):
+
+    def test_superuser(self):
+
+        db = get_user_model()
+
+        # Check if create_superuser creates a good superuser
+        super_user = db.objects.create_superuser(
+            'email@address.com', 'first_name', 'last_name', '1900-01-01', 'password'
+        )
+        self.assertEqual(super_user.email, 'email@address.com')
+        self.assertEqual(super_user.first_name, 'first_name')
+        self.assertEqual(super_user.last_name, 'last_name')
+        self.assertEqual(super_user.birth_date, '1900-01-01')
+        self.assertEqual(super_user.is_active, True)
+        self.assertEqual(super_user.is_staff, True)
+        self.assertEqual(super_user.is_superuser, True)
+        self.assertEqual(str(super_user), 'email@address.com')
+
+        # Check if email field is unique
+        with self.assertRaises(IntegrityError):
+            with transaction.atomic():
+                db.objects.create_superuser(
+                    'email@address.com', 'first_name', 'last_name', '1900-01-01', 'password'
+                )
+
+        # Check if create_superuser must have is_staff=True
+        with self.assertRaises(ValueError):
+            with transaction.atomic():
+                db.objects.create_superuser(
+                    'email_1@address.com', 'first_name', 'last_name', '1900-01-01', 'password', is_staff=False
+                )
+
+        # Check if create_superuser must have is_superuser=True
+        with self.assertRaises(ValueError):
+            with transaction.atomic():
+                db.objects.create_superuser(
+                    'email_2@address.com', 'first_name', 'last_name', '1900-01-01', 'password', is_superuser=False
+                )
+
+    def test_user(self):
```

Comment:
Usually, when you have a lot of assertions in a test, It's a sign that you need to divide it to separate test functions. 
instead of adding comments before each assertion, have a separate named test for each.

for example: 
```
test_email_not_unique_should_raise():
    with self.assertRaises(IntegrityError):
        with transaction.atomic():
            db.objects.create_user(
                'email@address.com', 'first_name', 'last_name', '1900-01-01', 'password'
            )
```

if some of the tests have something in common they need in order to work (like the user model), maybe it can be added as a fixture
https://docs.pytest.org/en/stable/fixture.html

Reply:
Another option can be using pytest's `parametrize` - this way you can run the same test with different parameters on each run.
in this case, the parameters can be the values you insert to the DB (either as a single parameter, or separately), and the specific error that should be raised. 

https://docs.pytest.org/en/stable/parametrize.html

----------------------------------------------------------------------
Both are very useful and powerful tools pytest provides, and you can write much better tests if you'll get used to them.